plugins/org.komodo.core/src/org/komodo/repository/ObjectImpl.java

```diff
@@ -541,6 +542,40 @@ public Descriptor getPrimaryType( final UnitOfWork uow ) throws KException {
         }
     }
 
+    protected <T> T getObjectProperty(UnitOfWork uow, Class<T> propertyType,
+                                                     String getterName, String propertyPath) throws KException {
+        UnitOfWork transaction = uow;
+
+        if (transaction == null) {
+            transaction = getRepository().createTransaction(getClass().getSimpleName() + HYPHEN + getterName, true, null);
+        }
+
+        assert (transaction != null);
+
+        if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug(getterName + ": transaction = '{0}'", transaction.getName()); //$NON-NLS-1$
+        }
+
+        try {
+            Property result = getProperty(transaction, propertyPath);
+
+            if (uow == null) {
+                transaction.commit();
+            }
+
+            if (result == null) {
+                return null;
+            }
+
+            if (String.class.equals(propertyType))
```

Comment:
I do prefer the more explicit methods but this will work. I know this is a `protected` method but I think I would add a javadoc explaining its limitation (only works for `String` type).